### PR TITLE
Fetch measures

### PR DESCRIPTION
**Pull request by eedrummer**
Implemented the ability to pull in the side bar using the new server at mitre/ecqm.

### CODE REVIEW

File: src/components/MeasureDisplay.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
 import React, { Component } from 'react';
+import { fetchMeasuresIfNeeded } from '../actions/index';
+import { connect } from 'react-redux';
+import MeasureCategory from './MeasureCategory';
+
+class MeasureDisplay extends Component {
+  constructor(props) {
+    super(props);
+  }
```
Comment (noranda) [inline src/components/MeasureDisplay.js]: Do we need this constructor if we're only calling `super`?

File: src/components/MeasureDisplay.js [hunk 2]
```diff
@@ -10,36 +21,12 @@ export default class MeasureDisplay extends Component {
...
-                  </div>
-                </div>
-              </div>
-            </div>
-          </div>
+          {this.props.categories.map(category =>
+            <MeasureCategory
+              category={category}
+              key={category}
+              measures={this.props.measures.filter((m) => {return m.category === category;})} />
```
Comment (noranda) [inline src/components/MeasureDisplay.js:24]: If you use the fat arrow without the curly braces it does an implicit return:

``` js
measures={this.props.measures.filter((m) => m.category === category)}
```

File: src/components/MeasureDisplay.js [hunk 3]
```diff
@@ -50,3 +37,50 @@ export default class MeasureDisplay extends Component {
...
+      dispatch(fetchMeasuresIfNeeded());
+    }
+  };
+};
+
+function flattenMeasures(measures) {
+  let uniqueMeasures = [];
+  measures.forEach((m) => {
+    var reducedMeasure = {cmsId: m.cmsId, name: m.name, category: m.category};
+    if (! uniqueMeasures.map((i) => { return i.cmsId; }).includes(reducedMeasure.cmsId)) {
```
Comment (noranda) [inline src/components/MeasureDisplay.js:49]: Same comment here about the fat arrow, you can skip the curly braces and the return in `map`:

``` js
uniqueMeasures.map((i) => i.cmsId)
```

**Comment by noranda** [comment]
Looks good :+1:

Needs some CSS tweaks but I can take care of that later.

**Comment by eedrummer** [comment]
I removed the unnecessary constructor on MeasureDisplay.

You are correct that we don't need the return keyword in the fat arrow functions. However, leaving it out makes JSHint raise a warning. I fiddled with the options, but I can't seem to find a way around it. Would you mind leaving them in for now? I like to keep the code JSHint clean. If you have other suggestions on how to configure this or use eslint in Atom, I'm all ears.

**Comment by noranda** [comment]
The project is set up to use eslint. Jshint doesn't support the jsx format, like eslint does (although there is jsxhint). I'll merge this, but we might want to talk about which linter to use going forward.

**Pull request merged**